### DISCUSSION TITLE
Unable to connect when using a DNS

### DISCUSSION THREAD

**Discussion by Marty42780** [cockpit-project/cockpit#17295] [Q&A] [answered]
### Explain what happens

1. When I connect in LAN (192.168.0.XX:9000), it works.
2. When I connect in WAN with a domain name, it refuses to login me.

I use the reverse DNS of my other NAS to connect to Cockpit and it automatically adds an SSL certificate

### Version of Cockpit

Version 239

### Where is the problem in Cockpit?

Networking

### Server operating system

Debian

### Server operating system version

Debian GNU/Linux 10 (buster) / OpenMediaVault server

### What browsers are you using?

Chrome

### System log

```shell
cockpit-tls[7690]: cockpit-tls: gnutls_handshake failed: A TLS fatal alert has been received.
```

**Comment by martinpitt** [comment]
> I use the reverse DNS of my other NAS to connect to Cockpit and it automatically adds an SSL certificate

What do you mean by this exactly? Does your NAS provide a reverse proxy which does TLS by itself? Then cockpit-tls should receive a plain HTTP connection, and not attempt a TLS handshake. It will most likely fail to login (as this scenario needs to be configured), but you should see the login page. So I'm afraid I need some more explanation what happens here.

**Comment by Marty42780** [comment]
Like you said, I have the login page but when I login I have an error.
I use the reverse DNS of my Synology DS to connect to my server (which have Cokpit) from WAN. This reverse DNS automatically add a TLS certificate. The reverse DNS send a http request to cockpit-tls.

**Comment by martinpitt** [comment, marked as answer]
Ah, that wasn't clear, thanks. For that to work you need to configure cockpit.conf to be aware of the proxy. See e.g. https://github.com/cockpit-project/cockpit/wiki/Proxying-Cockpit-over-NGINX or https://github.com/cockpit-project/cockpit/wiki/Proxying-Cockpit-over-Apache-with-LetsEncrypt how to do that.

**Reply by Marty42780** [reply]
I will see this thank you. 🙃
